stdlib/LinearAlgebra/src/lq.jl

```diff
@@ -333,3 +333,25 @@ function ldiv!(A::LQ{T}, B::StridedVecOrMat{T}) where T
     lmul!(adjoint(A.Q), ldiv!(LowerTriangular(A.L),B))
     return B
 end
+
+
+_tau(Q) = Q.τ
+_tau(Q::QRCompactWYQ) = @view Q.T[diagind(Q.T)]
```

Comment:
Did you do the math for this case? Great if it also works for blocks.

Reply:
Reading the docs https://docs.julialang.org/en/latest/stdlib/LinearAlgebra/#LinearAlgebra.QRCompactWY, I thought that I can just use the diagonal to extract the tau.  But that's actually not the case when `minimum(size(A)) > 36`?

According to LAPACK docs http://www.netlib.org/lapack/explore-html/dd/d9a/group__double_g_ecomputational_gaddcf152e87deec6123a1899f6f51101e.html#gaddcf152e87deec6123a1899f6f51101e, `F.T` is an `hcat`ed upper triangular matrices `T_n`, i.e.,

```
F.T == [T_1 T_2 ... T_B]
```

Does it mean

```
tau == [diag(T_1); diag(T_2); ...; diag(T_B)]
```

?